File: ROID-Forum-Server/ROID-Forum-Server/Convert Server Code/GMGServer/bin/Server/Sections/Game Section Controller/CommentInfo.dart
library GameCommentInfo;
import '../../Account Controller/AccountController.dart';

class CommentInfo {
  int threadID;
  int commentID;
  String comment;
  String owner;
  CommentInfo.fromData(int _threadID, int _commentID, String _comment, String _owner) {
    threadID = _threadID;
    commentID = _commentID;
    comment = _comment;
    owner = _owner;
  }
  CommentInfo.fromMap(Map m) {
    threadID = m["ThreadID"];
    commentID = m["CommentID"];
    comment = m['Comment'];
    owner = m['Owner'];
  }
  Map toMap({AccountController addAvatars: null}) {
      Map m = new Map();
      m["ThreadID"] = threadID;
      m["CommentID"] = commentID;
      m['Comment'] = comment;
      m['Owner'] = owner;
      if (addAvatars != null) {
        m['AvatarURL'] = addAvatars.getAvatarForAccount(owner);
      }
      return m;
    }
}

File: ROID-Forum-Server/ROID-Forum-Server/Section Controller/CommentInfo.cs
using System;
using System.Collections.Generic;
namespace ROIDForumServer
{
    public class CommentInfo
    {
        public int threadID;
        public int commentID;
        public String comment;
        public String owner;
        public String avatarURL;
        public CommentInfo()
        {
        }
        public CommentInfo(int _threadID, int _commentID, String _comment, String _owner, String _avatarURL)
        {
            threadID = _threadID;
            commentID = _commentID;
            comment = _comment;
            owner = _owner;
            avatarURL = _avatarURL;
        }
        public CommentInfo(Dictionary<string, object> m)
        {
            threadID = Convert.ToInt32(m["ThreadID"]);
            commentID = Convert.ToInt32(m["CommentID"]);
            comment = (string)m["Comment"];
            owner = (string)m["Owner"];
        }
        public Dictionary<string, object> toMap()
        {
            Dictionary<string, object> m = new Dictionary<string, object>();
            m["ThreadID"] = threadID;
            m["CommentID"] = commentID;
            m["Comment"] = comment;
            m["Owner"] = owner;
            m["AvatarURL"] = avatarURL;
            return m;
        }
    }
}